### ISSUE TITLE
Make armor useful for non-small ships

### ISSUE THREAD

**Issue by dahaic** [ospaceteam/outerspace#76]
HP has different "value", depending on the size of the ship. Destroying 100 HP of large ship is relatively easy, destroying 100 HP of small ship takes more effort and more weapons. Current implementation of armor technologies is subpar, because it's balanced around small ships, But using it on bigger ships is hugely cost ineffective.

This proposal is to introduce variants for all three sizes. By allowing only one hull size per variant, it shouldn't fill up the Construction dialog.

Proposed HP progression, 1x, 2x, 4x, but it might be necessary to alter it a bit, together with weight.

Names  of the new modules: <material> reinforcement, <material> armor plates, <material> bulkheads

**Comment by dahaic** [comment]
Analysis done with help of expression 
```
( DMG/t * HP ) / Price ^ 2
```
shown the small plates are not useful enough.

New HP proposed is 7 / 14 / 28. Not every design will benefit from plating, but most of them do. Then it is up to player and his speed preferences.
Higher TL armors should get a bit higher price, so their use is not without drawbacks.

**Comment by dahaic** [comment]
Because original plan was to unlock plates for all three sizes with one technology, this is currently blocked on update of research, as we don't have decoupled technologies and items.

**Comment by temuchin-** [comment]
I have a way to add an armor ,,, just for Large / Huge vessels. But before trying that out, I think the entire combat system should be examined to how it is meant to interact.

The system works like platemail, chainmail & leather armor  VS Pikes/ piercing, maces/ hammers, & bladed weapons. Each has an advantage that is offset by one other item. This is further balanced w/  Cost factors, time cost of training, & stealth vs sensors. All of which form the economics of Outer Space warfare. Further divided is the characteristics of each 'Base' Race (Later we might want to discuss the hybrids you can build in game thanks to the way the rules work). Borgs Bash deal lots of dmg & can have huge armored ships. Especially light ones. Mentats are good at all aspects & are balanced.  Bionics are the masters of economical warfare, aka Zurg Rush w/ some heavy skilled hitters.

**Comment by dahaic** [comment]
Combat system needs a rework, or at least a big update, that's for sure. But what needs to be said - it's probably not going to be by way of making even more rules, and making the system more complex. Because the game is complex as it is, and when you complicate stuff, you put new players in disadvantage by default. And that's not what we should strive for.

What I would like to move forward is to have about same amount of the options we have now, but maybe a bit more meaningful. And it needs to be simple to explain, and battles need to work in expected way. But at the same time, I would be wary of making it rock-paper-scissors, as you have to change battle logic so your fleets try to fight in optimal way. That's I would say very painful, and most likely outside what this project can handle :)

**Comment by temuchin-** [comment]
This is great we agree, it needs rework. But feel that it should be tackled later, as any optimizations should be done when the game is almost finished. Right now I am trying to find out how to make large ships useful in late game. This may be a situation where tactics & strategy needs to be modified to meet the game/combat system already in place. Will report back when I have found solutions / exhausted in game options.

**Comment by temuchin-** [comment]
Novel idea, add Damage Reduction (DR) to control race specific control modules. I have been testing this & it seems to work Also w/ the max DR of 5 (set in __init_py iirc) Changing the Damage for all weapons (anti med & anti large) is something I am running tests on. Will test large ships later this week w/ 1/2 dmg for anti large wpns.

Right now testing medium ships w/ DR & it seems a mild success

Suggested Values for now
Bridge 0
Living Bridge, Mind-Linked Bridge DR1
Computerized Bridge DR1 or DR0 (can't remember if DR from different sources stacks will be looking later when i pour through battle logs)
Command Deck DR2
Command Deck  Human, Bionic, Cyborg   ??DR3, 4, 5?? or maybe ?2, 3, 4? or maybe balanced? 3 to 4

Biometric Controller (umm pass, have used it for side by side variance tests but that will have to be high on the scale i am testing (it is supposed to be an elite controller for medium & large ships but it needs rework all over))

So far I have had success w/ 1/2 Dmg for anti large wpns, & 2/3 dmg for ant medium wpns.

This also has made it apparent that having a universal constant option would be great for the game & for dev work
Novel idea, add Damage Reduction (DR) to control race specific control modules. I have been testing this & it seems to work Also w/ the max DR of 5 (set in __init_py iirc) Changing the Damage for all weapons (anti med & anti large) is something I am running tests on. Will test large ships later this week w/ 1/2 dmg for anti large wpns.

Right now testing medium ships w/ DR & it seems a mild success

Suggested Values for now
Bridge 0
Living Bridge, Mind-Linked Bridge DR1
Computerized Bridge DR1 or DR0 (can't remember if DR from different sources stacks will be looking later when i pour through battle logs)
Command Deck DR2
Command Deck  Human, Bionic, Cyborg   ??DR3, 4, 5?? or maybe ?2, 3, 4? or maybe balanced? 3 to 4

Biometric Controller (umm pass, have used it for side by side variance tests but that will have to be high on the scale i am testing (it is supposed to be an elite controller for medium & large ships but it needs rework all over))

So far I have had success w/ 1/2 Dmg for anti large wpns, & 2/3 dmg for anti medium wpns.

**Comment by temuchin-** [comment]
Update on testing.
Parameters (same species vs same species, using general purpose ships)
(general purpose means about the same amount of wpns vs any target, w/ the baseline being 1 best anti-small, 1 best anti medium, 1 best anti large. 2 of each on medium 3 of each on large)
Humans vs Humans (note very great medium & large ships; this is their skew)
Cyborgs vs Cyborgs (make very good offense poor Def, Nano armor is a big factor; semi balanced.)
Bionics vs Bionics (great small ships, avg med, poor Large, Hull regen & numbers is a big factor)

Current testing is Humans vs Humans Large vs small (using small as yard stick for other ships)
Initial goal, to have the same # of Large ships not lose to the same # of small ships
Tools to work w/ Damage of wpn groups, ROF of wpn groups, built in Dmg Absorb in larger hulls.
Wpn Classes
small mounted anti-small; anti-med; anti-large; plus all of the above
Medium mounted anti-small; anti-med, anti-large; plus all of the above
Large mounted anti-small; plus all of the above
(Maybe (big maybe) add anti capital ship weapons mounted on capital ships?)

1st run has been reducing all ant-large wpns dmg to  1/2 & ant-medium wpns to 2/3.
100 large vs 100 small = complete loss for small ships. Large fleet lost ~30 ships.
25 large vs 100 small = draw (6 large ships left, no fighters)
15 large vs 100 small = draw (3 large ships left, no fighters)
10 large vs 100 small = minor victory for fighters, they lost ~40% while large lost 100%
5 large vs 100 small = total victory for fighters, they lost ~10%, large lost 100%.
While total victory is usually achieved w/ a 2 to 1 greater force in practice & in game; this means that 10 fighters would now be equivalent to 1 large ship.
Note <10 loss is considered total victory, Decisive victory is <20%, <40% is victory. Higher loss rates enter the realm of marginal victory territory >40% to <80%. Draw is @ >80% up to 100% loss. I could add in that technically anything less than 100 loss is victory making 80 to 99% loss a pyrhhic victory, & that would mean the same thing, that the force has a unsubstainable win, which is at best a draw.
Will post the rest of large fleet battles later.

**Comment by dahaic** [comment]
Hi,
first off, I have edited your previous comment, the body was repeated there twice. It should be ok now.

Thanks for the testing! When I was analyzing balance of the ships (or tried to), I have found out you cannot really compare ships of different speeds, as engines are very significant part of the ship cost. I would propose speed 5 or 6 as the right baseline (of course, races will have different balance on other speed levels). Another important part of medium/large ships balance are additional modules, like shields, ECMs. These either cannot be fitted on the small ship, or fitting them results in highly inefficient design, so doing balance through simple designs would actually make them overpowered.

**Comment by temuchin-** [comment]
----This post was a three day job/ culmination of weeks of research & applied theory, Also thanks for the edit above (i have been knee deep in stuff today)------

     Ok Just about run through all weapons & all empire types. Lowering the damage from fighters minHull="0" for anti med / lrg weapons weaponClass="1" weaponClass="2"

 minHull="0" weaponClass="1" = 2/3 damage (round up as needed)

 minHull="0" weaponClass="2" = 1/2 damage (round up as needed)

 minHull="1" weaponClass="1" = 2/3 damage (round up as needed)

 minHull="1" weaponClass="2" = 1/2 damage (round up as needed)

 minHull="2" weaponClass="1" = 100% damage

 minHull="2" weaponClass="2" = 100% damage

Also incorporating Titanium medium / large hulls; improved Regen for Medium/ Lrg Bio (3% & 4% respectively). Note testing assumes the following guidelines; only used general purpose ships (ships with about equal number/ weight of weapons alloted to anti small, anti med, anti large duty.) Also used max armor w/ speeds between 6 to 7; any relevant special equip; All ships are Level 4 for testing purposes; & I used the human general purpose fighter (AKA Hum Big Ftr) as the  base line unit vs all med / large ships (hum cyb bio).  SEE ATTACHED FILES.
     I added in special purpose ships (anti med, anti small, anti large) to get an idea how it can skew the numbers, plus testing to make sure it doesn't become more productive to just load a large ship w/ lots of anti small wpns to replace other systems, relying on chewing up the enemy fast enough from bottom to top.
     The goal was to have a Lrg ship be worth 10 small ships (which seems to translate into an advantage of about 2 to in 1 support cost, meaning for every 1 Construction Point you put into a large ship  >= to 2 CP put into a small ship); & Med ships worth 5 small ships (or 1.6 to 1 CP advantage over small ships).
Humans were spot on, cyborgs were above the threshold, & bio was below the threshold (awesome small ships while the med / large are subpar)


POST OPERATION NOTES
Increasing ship control hp could also have overall HP (my guess is between15 to 20 times base hp of bridge, command decks &  species variants)
=== The problem here is that it skews ship efficiency far beyond any armored ship variants, making armor even more useless on med / lrg ships while making med large ships an insurmountable threat.

The other option of implementing Dmg absorb into ship control might be workable,,,, but not something I feel worth testing @ the moment.

Increasing regen has very little effect on battles where the bio is not numerically superior, but helps recovery after battles so it does provide benefit.

 Also I have considered increasing medium ship wpn damage to 100%(vs small); 100%(vs med); 2/3 (vs Large)

My biggest struggle to overcome was the numerical superiority trap(with enough numbers quantity has a quality all its own, and this was baked into combat). The game considers >= 2times the number of equiv type ships the enemy has to be the tipping point for an overwhelming victory.

I also tested exotic strat res (huge hulls, mobile planet, fission drive, battle scanner, biometric controller, volley rockets, & a favorite plutonium mortar) w/ mixed results some items more than pay for the investment, others are pointless (super fuel tanks, battle scanner & biometric controller are the biggest failures). This is something I can address later.

     The final result is, the testing /tweaks does make it possible to put armor on med / lrg ships & get value about equal to their smaller brethren. If you run the attached files you will notice I added a hybrid option to my version, w/ less powerful pirates & all stratres, plus synthesis for all elements. The hybrid galaxy type was my first stab at such, I have tweaked that to work better for any future testing. The Elemental synthesis was solely for Development work, I did try to make synthesis expensive enough to be comparable. But shouldn't have tried, being able to synthesize the higher end elements in a regular game breaks so many things, Any thoughts to fix this should be put into the same place as addressing super strat res tech.

     In Conclusion (bringing this wall of text to an end); There are many ways to build workable fleet, which should mean there is no magic bullet design. Build w/ the idea of supportive general purpose ships, & tweak to meet the needs of the moment. Only live testing will test this. Because ships w/ little armor are still the most efficient, yet there are advantages to have a number of over armored ships an empire. This is as it should be in a strategy game that encompasses economics of warfare. This means Military Power (MP) is still a bad/ incomplete yardstick, which is also as it should be.

Plus the testing is not done,,,,, yet. I will post a condensed final notes (aka TL/DR;)

**Comment by temuchin-** [comment]
Wow just moved onto Large ship vs Large ship. Need to tweak some Min Hull 1 wpns. Suffice it to say; the Cyborgs are really happy w/ the results. I might need to add a save all ship config data to flat file option. Would make this a little easier. Also read in ship config from flat file option. Not looking at this for game purposes, just dev purposes.

**Comment by temuchin-** [comment]
Semi Weekly update. I have about balanced the use of armor.

***Short version:
 altering the wpn damages as noted above has led to large ships being a very worthwhile investment. Maybe shifting the damage mod one step further (from 1 ; 1/2; 1/4 to 1 , 1/4, 1/8) would help encourage larger ships while being a semi elegant solution. While altering defense values would be counterproductive, & time consuming.

****Long Version:
 Noting that Cyborgs get the best of weapons/ Armor/ Damage absorb/ Best multiplicative Atk mod (2 types of engine * psionic beacon)/ best efficiency in support costs (but worst fuel efficiency), this helps offset the lack of efficiency in buildings; especially in food & power (also industry tied to unalterable minerals, well unalterable for them). So keep in mind a decent cyborg Large ship has an attack of around 40!! while a top end human ship has an attack of 15, while bio is lucky to have an attack over 5. I am ignoring Medium ships & small ships at the moment; which have a much different split among the species.
>>>1st. Bio is left so far behind that the only way they ever would build large ships, is to keep the enemy honest, except they die so quickly to anti medium weapons that you would have to proportionately invest in medium shps as well. To address this I have experimented w/ scaling Live Large Hull w/ Titanium Large Hull improvements to Adv Large Hull. With some success.

>>>2nd. Bio race has no reason to invest in Titanium Large Hulls (or any titanium hull models). Not sure how this ever gets addresses outside of not offering them option of building titanium hulls (possibly replacing tech X w/ alternative Y) For the scope of this exercise I am only looking at Large hulls so please present any options only w/ that in mind. (I have looked at providing bio w/ armor X that requires uranium to build, can only be applied to medium / Large hulls gives armor like nanoweb w/ dmg absorb but no limit on total number, w/ mixed results)

>>>3rd. Only Bio & humans really gain advantage in Large ship combat by splitting out fleets into single components. Dmg Absorb is just that awesome for Cyborg (works all the time/ every shot). Whereas shields & regen only removes X damage at the start of the turn. This causes other problems...

>>>4th. The high values of cyborgs means they really don't want very much armor. As they can fire as many weapons as you can fit on a hull & all will hit. (~40 atk Large, ~ 50 atk med, ~60 atk small)

>>>5th. Humans get middle of the pack. Still one of the most interesting things is the final shield decision. Is your biggest opponent cyborg or bionic? That  leads to the biggest shift in the human endgame. So this helps play to the humans adaptability in all things design, which is evident throughout human empire / tech / buildings.

**Comment by temuchin-** [comment]
Updated previous post. To repeat we should not add damage absorb (sometimes i call this DR) to Regen hulls.

**Comment by temuchin-** [comment]
[Mod -formodifications.txt](https://github.com/ospaceteam/outerspace/files/2024572/Mod.-formodifications.txt)
Relevant files
[TL5.xml.txt](https://github.com/ospaceteam/outerspace/files/2024573/TL5.xml.txt)
[TL6.xml.txt](https://github.com/ospaceteam/outerspace/files/2024575/TL6.xml.txt)
[TL1.xml.txt](https://github.com/ospaceteam/outerspace/files/2024576/TL1.xml.txt)
[TL3.xml.txt](https://github.com/ospaceteam/outerspace/files/2024577/TL3.xml.txt)
[TL4.xml.txt](https://github.com/ospaceteam/outerspace/files/2024580/TL4.xml.txt)
[TL2.xml.txt](https://github.com/ospaceteam/outerspace/files/2024583/TL2.xml.txt)

**Comment by temuchin-** [comment]
Another idea in this department. Change the amount of space available. Right now making small ships that carry 3 to 4 bombs is fairly easy & as a bonus you can make them very useful (depending on tech base for L3). Not so for L4 Medium and L5 Large. You can Heavily armor them (about 1 to 1.5 K on Medium ships & 2 to 4K for Large ships). But not really put the same sorta fire power on them at least for anti planetary. So maybe we need to re balance ship sizes and payload? Increase Med / Large, or decrease small?

**Comment by temuchin-** [comment]
Annual update. I have attempted another approach. using partial powers to increase ship support costs. I am currently hovering around 1.6
i.e.
                obj.operProd += (tech.buildProd * Rules.operProdRatio)**1.6
So a 600 HP small ship is still possible, and I can build 140  efficient adv small hulls for the maintenance price of that ship, this will put brakes on the feed back loop of high armor ships and repair facilities (even after factoring in  experience modifiers and the ability to use hp as additional fuel tanks in a pinch)
Pros:
Helps remove micromanaging large HP fleets as the only viable path to combat victory. Players can now invest in cheap and efficient ships to attack and defend w/o seeing as much of a production waste.

Cons:
adds fractional powers to the servers standard computation cycles (always finding a way to eat up spare cpu cycles)

Next test
                obj.operProd += (tech.buildProd * Rules.operProdRatio)**(1.6 - (hullID/10))
This should reduce the armor penalty for larger ship hulls. Making them even more efficient. This will really help humans (already the defacto powerhouse), and greatly help Cyborgs who suffer mid game and boost their larger ships in the late game. And weirdly makes the cyborgs gravity surge mortars even more effective. Bionics suffer do to the effectiveness of their small hulls and because they never get ECM for their larger hulls, instead relying on regeneration. I have been experimenting on increasing the Regen to compensate for their larger hulls being bigger bullet magnets that will never dodge. Hopefully the next series of tests will yield more balanced results.

For reference I save a galaxy and run it manually and w/ a modified smoketest (ai preselects tech route and uses best hull, best control mod, best weapon X style, and best engine format to build ships, very rough old designs get left behind and only make upgrades after completing an entire odd #'ed research level 1 - 3 - 5). I also alternate between normal starts and delayed starts as a sort of stress test on the species type (I have yet to find an unwinnable scenario w/ humans, but have come very close).

**Comment by temuchin-** [comment]
Next test has hit a snag, when working out ship maintenance cost, the current calculation does not take into effect ship size. So before I break out this information, and then re-collate it. Will go to sleep on this problem.
